docs/reference/style-guides/node-style-guide.md

```diff
@@ -2,8 +2,186 @@
+```
+
+If a feature has many DTOs, models, etc., a hybrid approach may be used, organized by feature first:
+
+```
+app/
+├── auth/
+│   ├── dto/
+│   ├── model/
+│   |   └── auth.model.ts
+│   ├── auth.controller.ts
+│   └── auth.validator.ts
+```
+
+Tests should be organized alongside the code being tested, with the `.spec.ts` suffix:
+
+```
+app/
+├── auth/
+│   ├── auth.controller.ts
+│   ├── auth.controller.spec.ts
+```
+
+#### NestJS File Naming Conventions
+
+##### File Naming for Class
+
+```
+export class PascalCaseSuffix {} //= pascal-case.suffix.ts
+// Except for suffix, PascalCase to hyphen-case
+class FooBarNaming {} //= foo-bar.naming.ts
+class FooController {} //= foo.controller.ts
+class BarQueryDto {} //= bar-query.dto.ts
+```
+
+#### Interface Naming
+
+```
+// https://stackoverflow.com/questions/541912
+// https://stackoverflow.com/questions/2814805
+interface User {}
+interface CustomeUser extends User {}
+interface ThirdCustomeUser extends CustomeUser {}
+```
+
+#### Index Exporting
+
+```
+# It is recommended to place index.ts in each folder and export.
+# Unless it's a special case, it is import from a folder instead of directly from a file.
+- import { FooController } from './controllers/foo.controller';
+- import { BarController } from './controllers/bar.controller';
++ import { FooController, BarController } from './controllers';
+# My preferred method is to place only one fileOrFolder name at the end of the path.
+- import { UtilService } from '../common/providers/util.service';
++ import { UtilService } from '../common';
+```
+
+### TypeScript
+
+#### Tests
+
+All tests should be written in TypeScript.
+
+#### Never use Type Asserts
+
+Type asserts should not be used in any situation. Instead, use one of the following options:
+
+1. Check if the variable is null/undefined, and if so throw an error
+
+```typescript
+function example(arg: string | null) {
+  if (!arg) {
+    throw new Error('arg is required for example');
+  }
+
+  // arg is now of type 'string'
+}
+```
+
+2. Return early if it matches the desired behavior
+
+```typescript
+function example(arg?: string) {
+  if (!arg) return;
+
+  // arg is now of type 'string'
+}
+```
+
+3. Fallback to a default
+
+```typescript
+const DEFAULT_VALUE = 'example';
+
+function example(arg?: string) {
+  const argWithDefault = arg || DEFAULT_VALUE;
+
+  // argWithDefault is of type 'string'
+}
+```
+
+#### Use `satisfies` instead of `as`
+
+Prior to TypeScript 4.9, new literal objects could be identified as a type, but would not be narrowed. Our code uses TypeScript 4.9 or greater, so [`satisfies`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#the-satisfies-operator) should be preferred.
+
+## Legacy Code Packages
+
+### Style
+
+Legacy code has no defined style-guide beyond that enforced by ESLint.
+
+[ESLint] configuration is provided for each legacy package which enforces its requirements, it will be run by default during `git commit`.
```

Comment:
```suggestion
[ESLint][1] configuration is provided for each legacy package which enforces its requirements, it will be run by default during `git commit`.
```